client/js/components/pages/snippet-page.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PageWrapper from '../page-wrapper';
+import SnippetsList from '../snippets/snippets-list';
+import SearchBar from '../search-bar';
+import SnippetActions from '../../actions/snippet-actions';
+import SnippetStore from '../../stores/snippet-store';
+import UserStore from '../../stores/user-store';
+
+export default class SnippetPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = this.getPropsFromStores();
+    this._onChange = this._onChange.bind(this);
+    this.getSnippet = this.getSnippet.bind(this);
+  }
+
+  componentDidMount() {
+    this.storeListeners = [];
+    this.storeListeners.push(SnippetStore.listen(this._onChange));
+    this.getSnippet();
+  }
+
+  getSnippet(){
+    let foundSnippet = [];
+    let snippetId = this.props.params.id;
+    let currentUserId = this.getCurrentUser().id;
+    if(this.state.snippets.length == 0){
```

Comment:
I think something like this would look a little bit more readable:
```
0 >= this.state.snippets.filter(el => el.id === snippedId).length ? SnippetActions.getOneSnippet(this.props.params.id, currentUserId) : null;
```